mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte

```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+  import { Help } from '@mathesar-component-library';
+  import WarningBox from '@mathesar/components/message-boxes/WarningBox.svelte';
+  import type { UserModel } from '@mathesar/stores/users';
+  import { getSchemasStoreForDB } from '@mathesar/stores/schemas';
+  import { getUsersStoreFromContext } from '@mathesar/stores/users'; 
+  import type { Database, SchemaEntry } from '@mathesar/AppTypes';
+
+  export let database : Database;
+  export let schema : SchemaEntry;
+  export let userId: UserModel['id']
+  const usersStore = getUsersStoreFromContext();
+  $: user = usersStore?.getUserDetails(userId);
+  export let userWithRole : UserModel['schemaRoles'];
+  $: schemaStore = getSchemasStoreForDB(user?.databaseId);
+</script>
+
+<div class="user-permission-form">
+  <div>
+    <span>Permissions</span>
+    <Help>User Permissions</Help>
+  </div>
+  {#if !userWithRole }
+    <div class="user-without-permission">
+      <WarningBox fullWidth>
+        Please navigate to the appropriate database or schema in order to grant
+        access to the user.
+      </WarningBox>
+    </div>
+  {:else}
+  <div>
+    {database.name} / {schmema.name}
+  </div>

```

Comment:
The values here should be based on the permissions of the user and not the current database and schema. They can be obtained by `databaseRoles` and `schemaRoles` properties in the UserModel and the required API calls need to be made to obtain the names of the schemas and dbs we don't have yet.

Reply:
I am not able to get the schema name or database name from `schemaRoles` and `databaseRoles`. Could you please help me with how to make these api?

Reply:
@raunak-dev-edu Figuring these details out is a major part of implementing this requirement. You're free to find another issue to work on if you find it hard to implement this.